Refactor socks layer and enhance unittet

chhsiao90 opened the pull request:
Had refactor socks layer, please give me feedback if any, thanks!

chhsiao90 committed:
Refactor socks layer and enhance unittet

mike820324 commented:
Interesting PR, based on the diff it seems like you rewrite the socks layer.

mike820324 commented on microproxy/layer/proxy/socks.py:
```diff
@@ -33,120 +33,108 @@ def process_and_return_context(self):
     @gen.coroutine
-    def handle_request(self, event):
+    def handle_request_and_create_destination(self, event):
+        """Handle the socks requet from source
```
requet -> request

mike820324 commented on microproxy/layer/proxy/socks.py:
```diff
@@ -33,120 +33,108 @@ def process_and_return_context(self):
             logger.debug("Unsupport connect type")
-            error = ProtocolError("Unsupport bind type")
-            response_event = Response(
+            self.send_event_to_src_conn(Response(
```
Why there is no yield in here?

chhsiao90 replied:
Sorry, I forgot.

mike820324 commented on microproxy/layer/proxy/socks.py:
```diff
@@ -33,120 +33,108 @@ def process_and_return_context(self):
-                e, event, dest_stream)
-
+            dest_stream = yield self.create_dest_stream((event.addr, event.port))
+        except gen.TimeoutError as te:
```
is te and se really necessary? it seems the gen.Timeout and StreamClosedError are pretty self explain.

chhsiao90 replied:
Yeah~

chhsiao90 commented:
Updated and rebased

The major change of this PR is remove the return of tuple (error, event)
and let the handle_greeting_request , handle_request_and_create_destination to send event and raise Exception.

mike820324 commented:
Sorry for taking so long. LGTM now.
